src/lib/y2partitioner/widgets/disk_table.rb

```diff
@@ -0,0 +1,93 @@
+require "yast"
+
+require "cwm/table"
+
+require "y2partitioner/icons"
+require "y2partitioner/widgets/blk_devices_table"
+
+module Y2Partitioner
+  module Widgets
+    # Table widget to represent given list of Y2Storage::BlkDevice.
+    class DiskTable < CWM::Table
+      include BlkDevicesTable
+
+      # @param disks [Array<Y2Storage::Disk|Y2Storage::Partition>] devices to display
+      # @param pager [CWM::Pager] table have feature, that double click change content of pager
+      #   if someone do not need this feature, make it only optional
+      def initialize(disks, pager)
+        textdomain "storage"
+        @disks = disks
+        @pager = pager
+      end
+
+      # table items. See CWM::Table#items
+      def items
+        @disks.map do |device|
+          [
+            id_for_device(device), # use name as id
+            device.name,
+            device.size.to_human_string,
+            # TODO: dasd format use "X", check it
+            device.exists_in_probed? ? "" : _(BlkDevicesTable::FORMAT_FLAG),
+            encryption_value_for(device),
+            type_for(device),
+            fs_type_for(device),
+            device.filesystem_label || "",
+            device.filesystem_mountpoint || "",
+            device.region.start,
+            device.region.end
+          ]
+        end
+      end
+
+      # headers of table
+      def header
+        [
+          # TRANSLATORS: table header, Device is physical name of block device
+          # like partition or disk e.g. "/dev/sda1"
+          _("Device"),
+          # TRANSLATORS: table header, size of block device e.g. "8.00 GiB"
+          Right(_("Size")),
+          Center(_(BlkDevicesTable::FORMAT_FLAG)),
+          # TRANSLATORS: table header, flag if device is encrypted. Keep it short,
+          # ideally three letters. Keep in sync with Enc used later for format marker.
+          Center(_("Enc")),
+          # TRANSLATORS: table header, type of disk or partition. Can be longer. E.g. "Linux swap"
+          _("Type"),
+          # TRANSLATORS: table header, Files system type. Can be empty E.g. "BtrFS"
+          _("FS Type"),
+          # TRANSLATORS: table header, disk or partition label. Can be empty.
+          _("Label"),
+          # TRANSLATORS: table header, where is device mounted. Can be empty. E.g. "/" or "/home"
+          _("Mount Point"),
+          # TRANSLATORS: table header, which sector is the first one for device. E.g. "0"
+          Right(_("Start")),
+          # TRANSLATORS: table header, which sector is the the last for device. E.g. "126"
+          Right(_("End"))
+        ]
+      end
+
+      # @macro AW
+      def help
+        format(_(
+                 "Table shows selected devices with its attributes.<br>" \
```

Comment:
well, in general I think we should use more UX oriented way to proofread help. It should be read by our common user and say what he do not understand and what we should improve. And I think grammar is the last part